erts/emulator/beam/erl_bif_binary.c

```diff
@@ -2457,6 +2459,51 @@ static int cleanup_copy_bin_state(Binary *bp)
     return 1;
 }
 
+static BIF_RETTYPE binary_secure_compare(Process *p, Eterm bin1, Eterm bin2)
+{
+    byte *bytes1;
+    byte *bytes2;
+    Uint bitoffs1, bitsize1;
+    Uint bitoffs2, bitsize2;
+    Uint size1, size2;
+
+    size_t i;
+    volatile unsigned char acc = 0;
+  
+    if (is_not_binary(bin1) || is_not_binary(bin2))
+        goto bad_arg;
+    
+    ERTS_GET_BINARY_BYTES(bin1, bytes1, bitoffs1, bitsize1); 
+    ERTS_GET_BINARY_BYTES(bin2, bytes2, bitoffs2, bitsize2);
```

Comment:
This macro will do a realloc of the binary if it is an un-aligned sub binary, which means that it will take time depending on the size of the binary. Come to think about it, so will enif_inspect_binary.

Unaligned subbinaries is what `Bin` is if you do this:

```
<<_:1, Bin:1024/binary, _/bitstring>> = large_binary().
```

Not sure what do want to do when you get such a binary, either badarg or do some other trick.

Reply:
@garazdawi is the concern around performance or just general oddities that could happen as a result of unaligned subbinaries?

Reply:
The concern is that the realloc will take time proportional to the size of the binary. So if you want to have a constant time compare, you should think about this case.

Reply:
Gotcha. Let me do some spelunking.